package/package.mk

```
@@ -3,30 +3,41 @@ mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
 package_dir := $(notdir $(patsubst %/,%,$(dir $(mkfile_path))))
 
 crossplane_bin = $(go_bin)/kubectl-crossplane
+up_bin = $(go_bin)/up
 
 # Build kubectl-crossplane plugin
 $(crossplane_bin):export GOBIN = $(go_bin)
 $(crossplane_bin): | $(go_bin)
 	go install github.com/crossplane/crossplane/cmd/crank@latest
 	@mv $(go_bin)/crank $@
 
+# Install up plugin
+$(up_bin):export GOBIN = $(go_bin)
+$(up_bin): | $(go_bin)
+	curl -sL "https://cli.upbound.io" | sh
+	@mv up $@
+
 .PHONY: package
 package: ## All-in-one packaging and releasing
-package: package-push
+package: ghcr-package-push
 
 .PHONY: package-provider
 package-provider: export CONTROLLER_IMG = $(CONTAINER_IMG)
-package-provider: $(crossplane_bin) generate-go ## Build Crossplane package
+package-provider: $(up_bin) generate-go build-docker ## Build Crossplane package
 	@rm -rf package/*.xpkg
-	@yq e '.spec.controller.image=strenv(CONTROLLER_IMG)' $(package_dir)/crossplane.yaml.template > $(package_dir)/crossplane.yaml
-	@$(crossplane_bin) build provider -f $(package_dir)
+	@$(up_bin) xpkg build -f $(package_dir) -o $(package_dir)/package-exoscale.xpkg --controller=$(CONTROLLER_IMG)
 	@echo Package file: $$(ls $(package_dir)/*.xpkg)
 
-.PHONY: package-push
-package-push: pkg_file = $(shell ls $(package_dir)/*.xpkg)
-package-push: package-provider ## Push Crossplane package to container registry
-	$(crossplane_bin) push provider -f $(pkg_file) $(PACKAGE_IMG)
+.PHONY: ghcr-package-push
+ghcr-package-push: pkg_file = $(shell ls $(package_dir)/*.xpkg)
+ghcr-package-push: package-provider ## Push Crossplane package to ghcr container registry
+	$(crossplane_bin) push provider -f $(pkg_file) $(GHCR_PACKAGE_IMG)
+
+.PHONY: upbound-package-push
+upbound-package-push: pkg_file = $(shell ls $(package_dir)/*.xpkg)
+upbound-package-push: package-provider ## Push Crossplane package to upbound container registry
+	$(up_bin) xpkg push -f $(pkg_file) $(UPBOUND_PACKAGE_IMG)
```

I guess the individual targets aren't often used by developers, that's why I propose to make the "hidden" by prefixing with `.` to indicate an internal target, and to remove the comment so they don't show in `make help`
```suggestion
.PHONY: .ghcr-package-push
.ghcr-package-push: pkg_file = $(shell ls $(package_dir)/*.xpkg)
.ghcr-package-push: package-provider
	$(crossplane_bin) push provider -f $(pkg_file) $(GHCR_PACKAGE_IMG)

.PHONY: .upbound-package-push
.upbound-package-push: pkg_file = $(shell ls $(package_dir)/*.xpkg)
.upbound-package-push: package-provider
	$(up_bin) xpkg push -f $(pkg_file) $(UPBOUND_PACKAGE_IMG)
```

---

It's good that Docker login works on multiple repos at once. We can do what you proposed.